framework/web/Request.php

```diff
@@ -685,44 +697,55 @@ public function getBodyParams()
                 if (!($parser instanceof RequestParserInterface)) {
                     throw new InvalidConfigException("The '$contentType' request parser is invalid. It must implement the yii\\web\\RequestParserInterface.");
                 }
-                $this->_bodyParams = $parser->parse($this);
+                $this->_parsedBody = $parser->parse($this);
             } elseif (isset($this->parsers['*'])) {
                 $parser = Yii::createObject($this->parsers['*']);
                 if (!($parser instanceof RequestParserInterface)) {
                     throw new InvalidConfigException('The fallback request parser is invalid. It must implement the yii\\web\\RequestParserInterface.');
                 }
-                $this->_bodyParams = $parser->parse($this);
+                $this->_parsedBody = $parser->parse($this);
             } elseif ($this->getMethod() === 'POST') {
                 if ($contentType !== 'application/x-www-form-urlencoded' && $contentType !== 'multipart/form-data') {
                     throw new UnsupportedMediaTypeHttpException();
                 }
                 // PHP has already parsed the body so we have all params in $_POST
-                $this->_bodyParams = $_POST;
+                $this->_parsedBody = $_POST;
 
                 if ($contentType === 'multipart/form-data') {
-                    $this->_bodyParams = ArrayHelper::merge($this->_bodyParams, $this->getUploadedFiles());
+                    $this->_parsedBody = ArrayHelper::merge($this->_parsedBody, $this->getUploadedFiles());
                 }
             } else {
                 if ($contentType !== 'application/x-www-form-urlencoded') {
                     throw new UnsupportedMediaTypeHttpException();
                 }
-                $this->_bodyParams = [];
-                mb_parse_str($this->getBody()->__toString(), $this->_bodyParams);
+                $this->_parsedBody = [];
+                mb_parse_str($this->getBody()->__toString(), $this->_parsedBody);
             }
         }
 
-        return $this->_bodyParams;
+        return $this->_parsedBody;
     }
 
     /**
      * Sets the request body parameters.
      * @param array $values the request body parameters (name-value pairs)
-     * @see getBodyParam()
-     * @see getBodyParams()
+     * @see getParsedBodyParam()
+     * @see getParsedBody()
      */
-    public function setBodyParams($values)
+    public function setParsedBody($values)
```

Comment:
Do we still need to have `setParsedBody()` as public?

Reply:
I would rather keep it, since it was there before.
Besides having such public method can be useful while writing unit tests.

Reply:
> since it was there before.

That's not a good reason for a BC breaking release. As far as I understand PSR, whole idea is to make Request immutable i.e. no setters.